cfme/tests/infrastructure/test_host.py

```diff
@@ -430,3 +433,43 @@ def test_infrastructure_hosts_navigation_after_download(
     hosts_view.navigation.select("Compute")
     provider_view = provider.create_view(InfraProviderDetailsView)
     assert provider_view.is_displayed
+
+
+@test_requirements.infra_hosts
+@pytest.mark.meta(blockers=[BZ(1747545, forced_streams=["5.10"])], automates=[1747545])
+@pytest.mark.parametrize("hosts_collection", ["provider", "appliance"])
+@pytest.mark.parametrize(
+    "report_format", ["Download as Text", "Download as CSV", "Print or export as PDF"],
+    ids=["txt", "csv", "pdf"]
+)
+def test_infrastructure_hosts_navigation_after_download_from_compare(
+    appliance, setup_provider_min_hosts, provider, report_format, hosts_collection
+):
+    """
+    Polarion:
+        assignee: prichard
+        casecomponent: Infra
+        caseimportance: high
+        initialEstimate: 1/3h
+    Bugzilla:
+        1747545
+
+    """
+    if hosts_collection == "provider":
+        hosts_view = navigate_to(provider.collections.hosts, "All")
+    elif hosts_collection == "appliance":
+        hosts_view = navigate_to(appliance.collections.hosts, "All")
+    ent_slice = slice(0, 2, None)
+    num_hosts = hosts_view.entities.paginator.items_amount
+    if num_hosts < 2:
+        pytest.skip('not enough hosts in appliance UI to run test')
```

Comment:
I agree with @mshriver that this code block should be moved to a fixture to avoid code duplication. 

https://github.com/ManageIQ/integration_tests/pull/9472#discussion_r337157049